### ISSUE TITLE
Eliminate FLUTTER_NOLINT where possible once migrated away from RapidJson

### ISSUE THREAD

**Issue by cbracken** [flutter/flutter#65676]
Several of our message codecs and related files need `// FLUTTER_NOLINT` flags to disable clang linting due to issues with RapidJson. Example includes:

```
flutter/shell/platform/linux/fl_json_message_codec.cc:
../../third_party/rapidjson/include/rapidjson/internal/itoa.h:119:19: warning: Dereference of null pointer [clang-analyzer-core.NullDereference]
        *buffer++ = '-';
                  ^
/b/s/w/ir/cache/builder/src/flutter/shell/platform/linux/fl_json_message_codec.cc:249:8: note: Calling 'write_value'
  if (!write_value(writer, message, error)) {
       ^
/b/s/w/ir/cache/builder/src/flutter/shell/platform/linux/fl_json_message_codec.cc:28:7: note: Assuming the condition is false
  if (value == nullptr) {
      ^
/b/s/w/ir/cache/builder/src/flutter/shell/platform/linux/fl_json_message_codec.cc:28:3: note: Taking false branch
  if (value == nullptr) {
  ^
/b/s/w/ir/cache/builder/src/flutter/shell/platform/linux/fl_json_message_codec.cc:33:3: note: Control jumps to 'case FL_VALUE_TYPE_INT32_LIST:'  at line 58
  switch (fl_value_get_type(value)) {
  ^
/b/s/w/ir/cache/builder/src/flutter/shell/platform/linux/fl_json_message_codec.cc:61:26: note: Assuming the condition is true
      for (size_t i = 0; i < fl_value_get_length(value); i++) {
                         ^
/b/s/w/ir/cache/builder/src/flutter/shell/platform/linux/fl_json_message_codec.cc:61:7: note: Loop condition is true.  Entering loop body
      for (size_t i = 0; i < fl_value_get_length(value); i++) {
      ^
/b/s/w/ir/cache/builder/src/flutter/shell/platform/linux/fl_json_message_codec.cc:62:9: note: Calling 'Writer::Int'
        writer.Int(data[i]);
        ^
../../third_party/rapidjson/include/rapidjson/writer.h:183:72: note: Calling 'Writer::WriteInt'
    bool Int(int i)             { Prefix(kNumberType); return EndValue(WriteInt(i)); }
                                                                       ^
../../third_party/rapidjson/include/rapidjson/writer.h:515:23: note: Calling 'i32toa'
    const char* end = internal::i32toa(i, buffer);
                      ^
../../third_party/rapidjson/include/rapidjson/internal/itoa.h:118:9: note: Assuming 'value' is < 0
    if (value < 0) {
        ^
../../third_party/rapidjson/include/rapidjson/internal/itoa.h:118:5: note: Taking true branch
    if (value < 0) {
    ^
../../third_party/rapidjson/include/rapidjson/internal/itoa.h:119:10: note: Null pointer value stored to 'buffer'
        *buffer++ = '-';
         ^
../../third_party/rapidjson/include/rapidjson/internal/itoa.h:119:19: note: Dereference of null pointer
        *buffer++ = '-';
                  ^
../../third_party/rapidjson/include/rapidjson/internal/itoa.h:298:19: warning: Dereference of null pointer [clang-analyzer-core.NullDereference]
        *buffer++ = '-';
```

**Comment by cbracken** [comment]
Related #54693.

**Comment by cbracken** [comment]
Closed by flutter/engine#21921.
